paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetActivityTest.kt

```diff
@@ -33,27 +44,86 @@ internal class CustomerSheetActivityTest {
 
     @Test
     fun `Finish with cancel on back press`() {
-        val scenario = activityScenario().launchForResult(intent)
+        runActivityScenario { scenario ->
+            scenario.onActivity {
+                pressBack()
+            }
 
-        scenario.onActivity {
-            pressBack()
+            assertThat(
+                contract.parseResult(
+                    scenario.getResult().resultCode,
+                    scenario.getResult().resultData
+                )
+            ).isEqualTo(
+                InternalCustomerSheetResult.Canceled
+            )
         }
+    }
 
-        assertThat(
-            contract.parseResult(
-                scenario.getResult().resultCode,
-                scenario.getResult().resultData
+    @Test
+    fun `Finish with cancel on navigation icon press`() {
+        runActivityScenario(
+            viewState = CustomerSheetViewState.SelectPaymentMethod(
+                title = null
             )
-        ).isEqualTo(
-            InternalCustomerSheetResult.Canceled
-        )
+        ) { scenario ->
+            scenario.onActivity {
+                runTest {
+                    composeTestRule.waitUntil {
+                        runCatching {
+                            composeTestRule
+                                .onNodeWithTag(SHEET_NAVIGATION_BUTTON_TAG)
```

Comment:
Should we start building out a "Page" object for this?

It would basically just make the readability of tests like this more english like, and less technical.